store routing keys as public keys in did.

burdettadam opened the pull request:
bugfix for mediation connection protocol integration.

- `create_did_doc(...)` now stores routing keys as public keys objects instead of strings. when outbound transport consumes routing keys it requires a public key, not a string.
- code duplication is reduced by use of a decorator to check mediation parameters
- added mediation_record parameter to optimize decorator behavior
- small documentation changes

Signed-off-by: Adam Burdett <burdettadam@gmail.com>

dbluhm commented:
Spotted some issues; working with @burdettadam to address them. Specifically, using the decorator as it is now, it assumes that mediation_id and mediation_record are always passed as keyword arguments which doesn't match the method signatures decorated. Rather than modifying method signatures, I think we likely just need to simplify the validation into a plain old method.

burdettadam commented:
I wanted to see what it would take to get the decorator handling args and not just kwargs, so I added some tests and updated the decorator accordingly. I think the complexity of passing parameters into a decorator is not ideal. I will update this pr to reflect @dbluhm suggestions.

andrewwhitehead commented:
Looks pretty DRY at least!
